src/datepicker/datepicker.js

```diff
@@ -351,7 +351,7 @@ export default class Datepicker<T = Date> extends React.Component<
           if (displayValueAtRangeIndex === 0) {
             left = date;
             if (!right) {
-              onChange({date: [left]});
+              onChange({date: [left, null]});
```

Comment:
is setting this to null necessary, or is leaving [left, undefined] sufficient?

I've made similar comments regarding this idea in various places so far, and the point I want to make is that there seems to be an implicit difference between `undefined` and `null` values internally, and for consumers. It would be less complicated if we can operate as if undefined is essentially the same as null regarding the date value

Reply:
Yes, `undefined` and `null` are treated the same internally. My thinking in using `null` (even though not technically necessary) was that it was more explicit, but I get what you're saying and I totally agree. Using `null` in cases like the above makes it seem like there may actually be a difference between the two, which could be confusing for future devs reading the code. I'll review all the changes and make sure I'm not using `null` needlessly.